stacks-signer/src/stacks_client.rs

```diff
@@ -98,8 +142,157 @@ impl StacksClient {
+            .unwrap();
+        buf
+    }
 }
 
+impl InteractWithStacksContracts for StacksClient {
+    fn transaction_contract_call(
+        sender: &StacksPrivateKey,
+        nonce: u64,
+        contract_addr: &StacksAddress,
+        contract_name: &str,
+        function_name: &str,
+        function_args: &[ClarityValue],
+        tx_version: TransactionVersion,
+        chain_id: u32,
+    ) -> Vec<u8> {
+        let contract_name = ContractName::from(contract_name);
+        let function_name = ClarityName::from(function_name);
+
+        let payload = TransactionContractCall {
+            address: contract_addr.clone(),
+            contract_name,
+            function_name,
+            function_args: function_args.iter().map(|x| x.clone()).collect(),
+        };
+
+        let tx_fee = 0;
+
+        Self::seralize_sign_tx_anchor_mode_version(
+            payload.into(),
+            sender,
+            None,
+            nonce,
+            None,
+            tx_fee,
+            TransactionAnchorMode::OnChainOnly,
+            tx_version,
+            chain_id,
+        )
+    }
+
+    fn submit_tx(http_origin: &str, tx: &Vec<u8>) -> String {
+        let client = reqwest::blocking::Client::new();
+        let path = format!("{http_origin}/v2/transactions");
+        let res = client
+            .post(&path)
+            .header("Content-Type", "application/octet-stream")
+            .body(tx.clone())
+            .send()
+            .unwrap();
+        if res.status().is_success() {
+            let res: String = res.json().unwrap();
+            assert_eq!(
+                res,
+                StacksTransaction::consensus_deserialize(&mut &tx[..])
+                    .unwrap()
+                    .txid()
+                    .to_string()
+            );
+            return res;
+        } else {
+            // FIXME (https://github.com/stacks-network/stacks-blockchain/issues/3993): this needs to handled better
+            eprintln!("Submit tx error: {}", res.text().unwrap());
+            panic!("");
+        }
+    }
+
+    fn read_only_contract_call(
+        http_origin: &str,
+        sender: &StacksAddress,
+        contract_addr: &StacksAddress,
+        contract_name: &str,
+        function_name: &str,
+        function_args: &[ClarityValue],
+    ) -> String {
+        let body = json!({
+            "sender": sender.to_string(),
+            "arguments": function_args
+        })
+        .to_string();
+
+        let client = reqwest::blocking::Client::new();
+        let path = format!(
+            "{http_origin}/v2/contracts/call-read/{contract_addr}/{contract_name}/{function_name}"
+        );
+        let res = client
+            .post(&path)
+            .header("Content-TYpe", "application/json")
+            .body(body)
+            .send()
+            .unwrap();
+        if res.status().is_success() {
+            let res: String = res.json().unwrap();
+            return res;
```

Comment:
Mind running clippy on this PR? 
`cargo clippy -p stacks-signer --tests --no-deps --all-features`